packages/node_modules/@webex/plugin-meetings/src/locus-info/parser.js

```diff
@@ -1,118 +1,706 @@
+   */
+  static getNumbersOutOfRange(list, rangeStart, rangeEnd) {
+    const output = [];
+
+    list.forEach((num) => {
+      // if number is outside of specified range
+      if (num < rangeStart || num > rangeEnd) {
+        // add the number
+        output.push(num);
       }
-      break;
-    case DELTAEVENT.GT:
-      meeting.locusInfo.onDeltaLocus(deltaLocus);
-      // Turn off desync in case this delta locus comes from a sync request
-      meeting.locusDesync = false;
-      meeting.needToGetFullLocus = false;
-      break;
-    default:
-  }
-
-  return res;
-};
-
-
-Parser.syncMeeting = (meeting, deltaLocus) => {
-  if (meeting &&
-        meeting.locusInfo &&
-        meeting.locusInfo.syncUrl &&
-        meeting.locusInfo.fullState &&
-        meeting.locusInfo.fullState.state !== LOCUS.STATE.INACTIVE) {
-    // found that the locus syncs were happening even after 4 hrs of meeting
-    // completion, resulting in 403 errors. This check ensures we call
-    // syncing only if the locus state is not INACTIVE.
-    meeting.meetingRequest.syncMeeting({
-      syncUrl: deltaLocus.syncUrl,
-      desync: meeting.desync
-    })
-      .then((res) => {
-        meeting.locusInfo.onDeltaLocus(res.body);
-      });
-  }
-};
-
-export default Parser;
+    });
+
+    // sort ascending
+    return output.sort((a, b) => a - b);
+  }
+
+
+  /**
+   * Checks if newLoci or workingCopy is invalid.
+   * @param {Types~Locus} newLoci
+   * @returns {boolean}
+   */
+  isValidLocus(newLoci) {
+    let isValid = false;
+    const {IDLE} = Parser.status;
+    const {isLoci} = Parser;
+    const setStatus = function (status) { this.status = status; };
+
+    // nothing to do, queue is empty
+    if (!newLoci) {
+      setStatus(IDLE);
+    }
+
+    // one or both objects are not locus delta events
+    if (!isLoci(this.workingCopy) || !isLoci(newLoci)) {
+      setStatus(IDLE);
+      LoggerProxy.logger.info('Locus-info:parser#processDeltaEvent --> Ignoring non-locus object. workingCopy:', this.workingCopy, 'newLoci:', newLoci);
+    }
+    else {
+      isValid = true;
+    }
+
+    return isValid;
+  }
+
+
+  /**
+   * Determines if a paricular locus's sequence is empty
+   * @param {Types~Locus} locus
+   * @returns {bool}
+   */
+  static isSequenceEmpty(locus) {
+    const {sequence} = locus;
+    const hasEmptyEntries = !sequence.entries || !sequence.entries.length;
+    const hasEmptyRange = sequence.rangeStart === 0 && sequence.rangeEnd === 0;
+
+    return hasEmptyEntries && hasEmptyRange;
+  }
+
+
+  /**
+   * Determines if an object has basic
+   * structure of a locus object.
+   * @param {Types~Locus} loci
+   * @returns {boolean}
+   */
+  static isLoci(loci) {
+    if (!loci || !loci.sequence) {
+      return false;
+    }
+    const hasProp = (prop) =>
+      Object.prototype.hasOwnProperty.call(loci.sequence, prop);
+
+    if (hasProp('rangeStart') && hasProp('rangeEnd')) {
+      return true;
+    }
+
+    return false;
+  }
```

Comment:
The current function always returns a boolean, the suggested change returns `undefined` in certain instances. 

Also I tried to write very clear, often verbose code so that the intention is very apparent to the next dev. If someone wanted to add additional checks for what is a Loci (say Locus team changed structure of a Loci), I feel the current function makes it clear and obvious. 

The PR this replaces was very condensed, I intentionally tried to do the opposite.